### DISCUSSION TITLE
VDH suddenly not downloading correctly, often only 1kb & not giving me options, plus got "side download" error message when I wasn't doing a side download

### DISCUSSION THREAD

**Discussion by Anoneemouse** [aclap-dev/video-downloadhelper#85] [Help and Q&A]
### Your question

Hi,

I'm set to use the browser for downloads, and I'm using the current version of chrome on a windows 10 machine. Normally when I click on the active VDH icon, then select the file I want to download, it pops up a dialogue box allowing me to both name the file and specify where I want it to be downloaded to.  

Suddenly the last day or two none of that happens.  It defaults to the folder I have set for companion app downloads... and files that are anywhere from 250mb to 1G or more are often only downloading as 1kb - e.g., I can't get them at all.  Sometimes the full file is downloaded, but again, it's going into the folder I specified for HLS streams (e.g., when the companion app must be used) - and it's not giving me the option before downloading to name the file either.  I also had one that gave me an error message - something about an error with a side download.  

But my VDH is set to use the browser - and I haven't changed any of the settings.  I've never tried doing a side download.  

Please help me get it back working like it used to!!

### Addon & CoApp version and details

version: 8.2.0.20
channel: stable
build date: 2024-02-12
build options: linuxlic: false, noyt: true, browser: chrome
lang: en-US
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.10",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\noney",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "6.0"
}

**Comment by paulrouget** [comment]
Version 8.2.0.21 and above should fix your issue. Please be patient as the Chrome Webstore updates the addon.

**Reply by Anoneemouse** [reply]
Ok, thank you Paul - and thanks so much for the reply!!  I hope they update it soon - VDH has always been so useful - I love it!

**Reply by Anoneemouse** [reply]
@paulrouget 

Hi Paul, VDH now seems to be downloading mostly properly again - at least now I get the popup window asking me what I want to name the file and where I want it to be saved again, so that's good!  The only problem I see now is that it's not keeping the title of the file that's on it from the source in the popup window (e.g., where I can modify it further if desired)... it just has "v.mp4" for a title (which I can then modify).  I went to VDH options and double checked that it's set to keep long titles & "Download using the same headers as the original request."  It REALLY helps to have the hit/file title from the source, however, as that saves a lot of typing and often includes some file type details such as h264 or x265 etc.

Is there something else I can do to get it to offer me the full hit title in the save dialogue box that way? Or is this something that still needs to be fixed in VDH on your side?

Thanks in advance for your help on this!

Again, I'm using the most current version of chrome on a Windows 10 machine.  I've copied in the current VDH app information below so you can see which version it's on now, since it's updated from when I initially posted this question.  

version: 8.2.0.24
channel: stable
build date: 2024-02-16
build options: linuxlic: false, noyt: true, browser: chrome
lang: en-US
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.10",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\noney",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "6.0"
}

**Reply by GrampaWildWilly** [reply]
![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/4470ef5f-9926-4f17-9fbe-7f9d8e557e60)

**Reply by GrampaWildWilly** [reply]
![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/a34945b9-a788-4a67-94bf-ca60a70aecbe)

**Reply by Anoneemouse** [reply]
Hi GranpaWildWilly,

Thanks for the reply!  The line you're showing isn't on mine at all.  Darned if I can find a way to get a full capture of the window for you either - I can only get the visible part of the page (or I'd've posted it for you originally).  Here's what I see for just that part tho, if it helps.

![VDH screen grab](https://github.com/aclap-dev/video-downloadhelper/assets/16356703/0abc43c8-1e10-46c5-902a-389cd6bc9377)

For what it's worth, prior to the upgrade that broke things, this worked just fine as it is, and it'd display the file title from the source (and that I could modify before saving).  Now, at least for the two downloads I've tried, it just puts in a "v" with the file extension after that.  

Oh shoot, now I tried a third one, and it DID keep the file name... so now I'm not sure if it's working intermittently, or if maybe the first two were messed up from the source rather than VDH, or if perhaps you folks made another change to VDH that fixed it.  

Ok, I tried several other files - it seems intermittent, mostly either showing the "v" and file extension, or a bunch of letters and numbers and the file extension.  And when downloading the only one that DID show the right file name the first time, the second time I tried downloading it on that same file it didn't show the file name again - just a bunch of letters and numbers instead.

Here's an example of the save window where it just gave me the "v":

![VDH save from streamplay](https://github.com/aclap-dev/video-downloadhelper/assets/16356703/9d3b41ed-eebf-4198-ab55-6bbb82c65f35)

Here's one where it showed numbers and letters:

![VDH save from mixdrop link gives all characters](https://github.com/aclap-dev/video-downloadhelper/assets/16356703/e6f95e3d-e84b-4eac-a0f9-0823c4a9a30f)

Also interesting, one time when I selected a hit to download, but then canceled rather than saving from the save as window, I got a pop up message from the browser that the file was ready (even tho I hadn't downloaded anything). A couple of other times when I cancelled out of the save window it did NOT give me a message like that. So that was odd, but no big deal whether it does that or not for me at least.  Here's the message I got:

![VDH cancel download from the save window](https://github.com/aclap-dev/video-downloadhelper/assets/16356703/fd0f8e07-a223-4574-a918-3bb575b2a4f6)

**Reply by Anoneemouse** [reply]
And instead of the "v" or letters & numbers, it should be showing something like this (this is displayed when I hover over the top of the video screen, and normally it's the same thing that would show in the VDH save window, for me to keep or modify before saving):

![VDH example of file title](https://github.com/aclap-dev/video-downloadhelper/assets/16356703/f9e61f6e-15e0-47c8-86cf-3a8eae8abc4a)

**Reply by GrampaWildWilly** [reply]
The filepicker setting is available in recent betas on Firefox.  I'm not sure what the state of affairs is on Chrome.  In the typical situation, the changes in the Firefox betas eventually migrate to the other browsers.  If you're having problems that are keeping you on backlevel versions, you won't get the new features.  It's a tradeoff.  You'll have to weigh your options.

**Reply by Anoneemouse** [reply]
GrampaWildWilly, the old version prior to the one that broke VDH used to work just fine in this regard.  Is there any way to migrate back to it???  I'm surprised that it doesn't work in the latest version, and really hope you folks can get it fixed.  Tho I'm VERY glad that at least now I seem to be able to get downloads again, so the VDH folks have fixed some of the problems that update caused.  

But it isn't much of an option to have to switch to a Firefox beta, or just not have it no longer pick up the file name if we stay with chrome, unfortunately.  Plus, for me, I have to keep an old version of firefox for other purposes - so to even put the newer version of firefox on my computer, I have to figure out how to have two different versions with two different profiles on the same machine and do it carefully enough to not accidently mess up the old version of firefox that I need... and that's a pain in the butt to figure out for someone who's only done it once years ago (and found out in the process just how easy it is to screw up the old version, and then have to figure out some way to restore it, etc.).

**Reply by Anoneemouse** [reply]
@GrampaWildWilly ,  I thought the checkbox for "download using the same headers as the original request" was supposed to make VDH use the file name from the source, isn't it?  What's the difference between that and the checkbox you noted for "use the OS filepicker"?

**Reply by GrampaWildWilly** [reply]
I don't really know.  Remember, I don't work for VDH.  I'm just another user, like you.  I do know that the filepicker setting uses a different file selection dialog.  Once you get to that release, you'll see that it's the same as the file selection dialog you get in most standard applications.  It looks like the standard Windows 11 File Explorer, which is Microsoft's built in file manager.  The one you are stuck with on the release you are on is a file selection dialog that Michel concocted years ago.  I don't know what the other setting means.  I've never changed it from its default, which is that the box is checked.

**Comment by GrampaWildWilly** [comment]
For future reference, you ought to save yourself a lot of rather confusing verbiage & post screenshots instead.  Plus you should include the URL of whatever it is that isn't downloading for you.  That is, post the URL if anybody can access it without logging onto something.  In the case of needing to log on, your screenshots will substitute just fine for the rest of us not being able to access the content.

**Reply by Anoneemouse** [reply]
It's impossible to post screenshots of things that no longer occur, Grampa.  What could I possibly have screenshotted?  That a file which should have been hundreds of megabites was only 1kb?  There was nothing to screenshot.

**Reply by GrampaWildWilly** [reply]
I did start my comment with "for future reference."  I meant in future, you should consider posting a problem report that goes something like this:

I'm having a problem on this page:

www.somesite/something/something/whatever

When I visit the page, VDH tells me this:

screenshot of web page with the VDH menu open.  Depending on how many variants VDH offers, you might consider enough screenshots to show scrolling of the VDH menu.  You might use your image editing application to draw an arrow on the image to point to the variant you download.

I have Download (not Quick Download) set as my default action in VDH so I just clicked that variant.

screenshots here of the file selection steps.  Possibly not relevant.  You'd have to use your judgement here.

VDH downloaded that.

screenshot of what you get when you click the blue dot on the VDH menu while a download is in progress.  This would show the VDH status menu of the inflight download.

The download completed & I had these results:

screenshot of your file manager open on your target directory in details view with your downloaded file selected.  This would show file name, file size, last update date/time, possibly also resolution & frame rate, depending on what you told Windows to display.  And if you're not running Windows, show the analogous information from your platform.

screenshot of file properties dialog (another Windows thing).  The first tab of the dialog shows file size & the creation & last update time stamps, from which we can calculate the download time.  Another tab of the dialog shows video properties like resolution, frame rate, & bit rates, also shows audio properties like sampling rate & bit rate.

But when I go to play the file in VLC, it looks like this:

screenshot of the VLC window showing whatever is wrong.

That's what I'm suggesting.  For future reference.  Adjust this to suit the particulars of the case in question.

**Reply by Anoneemouse** [reply]
Ok.  That'd take a heck of a lot of time and effort to put together, getting a zillion screenshots and so on, opening & using a photo editor to edit them, uploading them into a bug report in the right order, etc.  And I sure don't see how a screenshot of the file manager with a 1kb downloaded file helps anything.  Mine isn't set to show a file's resolution and framerate - and how does that help on a 1kb file anyhow?  Or a screenshot of the VDH download drop menu either, when there are only one or two things on it and it's not the problem. 

If you folks ever need additional information that way, I'd sure try to get it if I could, but to file a bug report for something where a bunch of screenshots would tell you almost nothing, don't help explain the problem any more than a few words, but take a lot of time and effort to get, then upload back into the report and so on???  I sure don't see the point in doing that.  

If I'd thought screenshots would have been helpful, I'd've gotten them, and will be more than happy to do so in the future whenever it seems worth doing, or if someone replying asks for specific ones showing specific things. 

Unfortunately from other people's comments in the webstore, it looks like VDH still is broken, even in newer versions.  I haven't tried it again yet myself.
